### ISSUE TITLE
RallyRESTAPIError: the current Workspace |My Workspace| does not contain a Project that matches the current setting of the Project: None

### ISSUE THREAD

**Issue by noadmin** [RallyTools/RallyRestToolkitForPython#91]
After creating Rally instance, project var is passed as None to the final request:

``` python
from pyral import Rally
apikey = 'my_API_key'
wsp = 'My Workspace'
prj = 'existed project in workspace'
rl = Rally('rally1.rallydev.com', apikey=api_key, workspace=wsp, project=prj)

----> 1 rl = Rally('rally1.rallydev.com', apikey=api_key, workspace=wsp, project=prj)

/home/developer/.virtualenvs/s3venv/lib/python2.7/site-packages/pyral/restapi.pyc in __init__(self, server, user, password, apikey, version, warn, server_ping, isolated_workspace, **kwargs)
    244         if 'project' in kwargs and kwargs['project'] and kwargs['project']!= 'default':
    245             proj = kwargs['project']
--> 246         self.contextHelper.check(self.server, wksp, proj, self.isolated_workspace)
    247 
    248         if self.contextHelper.currentContext() not in _rallyCache:

/home/developer/.virtualenvs/s3venv/lib/python2.7/site-packages/pyral/context.pyc in check(self, server, workspace, project, isolated_workspace)
    189             self._getSubscriptionWorkspaces(subscription, limit=0)
    190         self._getWorkspacesAndProjects(workspace=self._currentWorkspace, project=self._defaultProject)
--> 191         self._setOperatingContext(project)
    192         schema_info = self.agent.getSchemaInfo(self._currentWorkspace)
    193         processSchemaInfo(self.getWorkspace(), schema_info)

/home/developer/.virtualenvs/s3venv/lib/python2.7/site-packages/pyral/context.pyc in _setOperatingContext(self, project_name)
    294             a match in the returned set raise an Exception stating that fact.
    295         """
--> 296         result = self.agent.get('Project', fetch="Name", workspace=self._currentWorkspace)
    297 
    298         if not result or result.resultCount == 0:

/home/developer/.virtualenvs/s3venv/lib/python2.7/site-packages/pyral/restapi.pyc in get(self, entity, fetch, query, order, **kwargs)
    900                 projectScopeDown=True/False
    901         """
--> 902         context, resource, full_resource_url, limit = self._buildRequest(entity, fetch, query, order, kwargs)
    903 
    904         if self._log:

/home/developer/.virtualenvs/s3venv/lib/python2.7/site-packages/pyral/restapi.pyc in _buildRequest(self, entity, fetch, query, order, kwargs)
    796             context = RallyContext(self.server, self.user, self.password or self.apikey, self.service_url)
    797         else:
--> 798             context, augments = self.contextHelper.identifyContext(**kwargs)
    799             workspace_ref = self.contextHelper.currentWorkspaceRef()
    800             project_ref   = self.contextHelper.currentProjectRef()

/home/developer/.virtualenvs/s3venv/lib/python2.7/site-packages/pyral/context.pyc in identifyContext(self, **kwargs)
    716         if self._currentProject not in self._projects[self._currentWorkspace]:
    717             problem = "the current Workspace |%s| does not contain a Project that matches the current setting of the Project: %s" % (self._currentWorkspace, self._currentProject)
--> 718             raise RallyRESTAPIError(problem)
    719 ##
    720         return self.context, augments

RallyRESTAPIError: the current Workspace |My Workspace| does not contain a Project that matches the current setting of the Project: None
```

**Comment by ClockerZadq** [comment]
Hey I had this same issue and posted it, its under issue 88. I also included a simple fix - I think the maintainer will be looking into this.

**Comment by klehman-rally** [comment]
This should be addressed by 1.2.1.  One thing to be aware of in 1.2.1, is that obtaining a pyral Rally instance now requires that you either have a default Workspace and Project for which your credentials entitle visibility OR that you explicitly supply at instantiation time a valid Workspace and valid Project to which your credentials entitle visibility.
